### PR TITLE
New resource: aws_neptune_cluster_instance

### PR DESCRIPTION
**Pull request by saravanan30erd**
Reference #4713 

Changes proposed in this pull request:

* Change 1
Added new resource aws_neptune_cluster_instance

### CODE REVIEW

File: aws/resource_aws_neptune_cluster_instance.go [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+	if err != nil {
+		if isAWSErr(err, neptune.ErrCodeDBInstanceNotFoundFault, "") {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("Error retrieving Neptune Instances: %s", err)
+	}
+
+	if len(resp.DBInstances) != 1 ||
+		aws.StringValue(resp.DBInstances[0].DBInstanceIdentifier) != id {
+		if err != nil {
```
Comment (bflad) [inline aws/resource_aws_neptune_cluster_instance.go]: Anytime `err != nil`, it is previously `return`ed -- this conditional seems like it should be removed.

File: aws/resource_aws_neptune_cluster_instance.go [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+				ValidateFunc: validateOnceAWeekWindowFormat,
+			},
+
+			"promotion_tier": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  0,
+			},
+
+			"publicly_accessible": {
```
Comment (bflad) [inline aws/resource_aws_neptune_cluster_instance.go]: Is this attribute updatable? It is not handled by the update function currently and should either be added there or have `ForceNew: true` added

File: aws/resource_aws_neptune_cluster_instance.go [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+		req.DBParameterGroupName = aws.String(d.Get("neptune_parameter_group_name").(string))
+		requestUpdate = true
+	}
+
+	if d.HasChange("instance_class") {
+		req.DBInstanceClass = aws.String(d.Get("instance_class").(string))
+		requestUpdate = true
+	}
+
+	if d.HasChange("monitoring_role_arn") {
```
Comment (bflad) [inline aws/resource_aws_neptune_cluster_instance.go]: We might want to add an acceptance test to verify that both the `MonitoringRoleArn` and `MonitoringInterval` parameters do not need to specified at the same time during update. See also: #2188

Comment (saravanan30erd) [reply aws/resource_aws_neptune_cluster_instance.go]: @bflad Looks like there is no support for enhanced monitoring for neptune engine 👍 

`aws_neptune_cluster_instance.cluster_instances: error creating Neptune Instance: InvalidParameterCombination: Enhanced Monitoring is not supported for the neptune database engine. Go to the RDS documentation for information about supported database engines.`

**Comment by saravanan30erd** [comment]
@bflad done all the changes

File: aws/resource_aws_neptune_cluster_instance.go [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+			if aws.BoolValue(m.IsClusterWriter) == true {
+				d.Set("writer", true)
+			} else {
+				d.Set("writer", false)
+			}
+		}
+	}
+
+	if db.Endpoint != nil {
+		d.Set("endpoint", db.Endpoint.Address)
```
Comment (bflad) [inline aws/resource_aws_neptune_cluster_instance.go]: Nitpick: with the `aws_db_instance` resource we have 3 attributes (`address`, `endpoint`, and `port`):

https://github.com/terraform-providers/terraform-provider-aws/blob/2c5c645f5a088c38ac490f032adaf1706e4c09a7/aws/resource_aws_db_instance.go#L987-L995

For a few reasons I think we should follow suit here:
* Provides an attribute with only the hostname
* Matches the API naming for address
* Ancillary: Reduces any confusion working between the two resources

File: aws/resource_aws_neptune_cluster_instance.go [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsNeptuneClusterInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNeptuneClusterInstanceCreate,
+		Read:   resourceAwsNeptuneClusterInstanceRead,
+		Update: resourceAwsNeptuneClusterInstanceUpdate,
+		Delete: resourceAwsNeptuneClusterInstanceDelete,
+		Importer: &schema.ResourceImporter{
```
Comment (bflad) [inline aws/resource_aws_neptune_cluster_instance.go]: Resource import documentation is missing

**Comment by bflad** [comment]
This has been released in [version 1.30.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1300-august-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**